src/idl_parser.cpp

```diff
@@ -555,7 +556,27 @@ CheckedError Parser::ParseType(Type &type) {
         return Error(
               "vector of union types not supported (wrap in table first).");
       }
-      type = Type(BASE_TYPE_VECTOR, subtype.struct_def, subtype.enum_def);
+      if (token_ == ':') {
+        if (!IsScalar(subtype.base_type)) {
+          return Error(
+                "arrays of non-scalar types not supported.");
+        }
+        NEXT();
+        if (token_ != kTokenIntegerConstant) {
```

Comment:
Though I guess that would need support in the lexical analyzer or type parser, so maybe not.
Maybe `field : [int 4]` would be better, to make it similar to vectors and clear it is an array.